packages/react/src/components/FocusTrapZone/FocusTrapZone.tsx

```diff
@@ -7,82 +7,78 @@ import {
 import { useDocument } from '../../WindowProvider';
 import type { IRefObject } from '../../Utilities';
 import type { IFocusTrapZoneProps, IFocusTrapZone } from './FocusTrapZone.types';
 
 interface IFocusTrapZoneInternalState {
-  disposeFocusHandler: (() => void) | undefined;
-  disposeClickHandler: (() => void) | undefined;
-  previouslyFocusedElementOutsideTrapZone: HTMLElement | undefined;
-  previouslyFocusedElementInTrapZone: HTMLElement | undefined;
+  previouslyFocusedElementInTrapZone?: HTMLElement;
   hasFocus: boolean;
-  unmodalize: (() => void) | undefined;
+  /** ID tracked in focusStack. Don't respect prop updates in case the ID changes while the FTZ is active. */
+  focusStackId: string;
 }
 
 const COMPONENT_NAME = 'FocusTrapZone';
 
-const useComponentRef = (
-  componentRef: IRefObject<IFocusTrapZone> | undefined,
-  previouslyFocusedElement: HTMLElement | undefined,
-  focus: () => void,
-) => {
+const DEFAULT_PROPS = {
+  disabled: false,
+  disableFirstFocus: false,
+  forceFocusInsideTrap: true,
+  isClickableOutsideFocusTrap: false,
+};
+
+const useComponentRef = (componentRef: IRefObject<IFocusTrapZone> | undefined, focusFTZ: () => void) => {
   React.useImperativeHandle(
     componentRef,
     () => ({
-      get previouslyFocusedElement() {
-        return previouslyFocusedElement;
-      },
-      focus,
+      focus: focusFTZ,
     }),
-    [previouslyFocusedElement, focus],
+    [focusFTZ],
   );
 };
 
 export const FocusTrapZone: React.FunctionComponent<IFocusTrapZoneProps> & {
-  focusStack: string[];
-} = React.forwardRef<HTMLElement, IFocusTrapZoneProps>((props, ref) => {
+  /**
+   * Stack of active FocusTrapZone identifiers, exposed for testing purposes only.
+   * (This is always set, just marked as optional to avoid a cast in the component definition.)
+   * @internal
+   */
+  focusStack?: string[];
+} = React.forwardRef<HTMLDivElement, IFocusTrapZoneProps>((propsWithoutDefaults, ref) => {
   const root = React.useRef<HTMLDivElement>(null);
   const firstBumper = React.useRef<HTMLDivElement>(null);
   const lastBumper = React.useRef<HTMLDivElement>(null);
   const mergedRootRef = useMergedRefs(root, ref) as React.Ref<HTMLDivElement>;
-  const id = useId(undefined, props.id);
   const doc = useDocument();
-  const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(props, divProperties);
 
-  const internalState = useConst<IFocusTrapZoneInternalState>(() => ({
-    previouslyFocusedElementOutsideTrapZone: undefined,
-    previouslyFocusedElementInTrapZone: undefined,
-    disposeFocusHandler: undefined,
-    disposeClickHandler: undefined,
+  const isFirstRender = usePrevious(false) ?? true;
+
+  const props = getPropsWithDefaults(DEFAULT_PROPS, propsWithoutDefaults);
+
+  const internalState = useConst<IFocusTrapZoneInternalState>({
     hasFocus: false,
-    unmodalize: undefined,
-  }));
+    focusStackId: useId('ftz-', props.id),
```

Comment:
The ID for focus stack purposes isn't directly used to find anything in the DOM, so it's safer to save it with `useId` to ensure the focus stack doesn't get messed up if `props.id` changes (which is not likely, but still)